pkg/virt-api/api.go

```diff
@@ -112,18 +305,245 @@ func addInfoToSwaggerObject(swo *openapispec.Swagger) {
 	}
 }
 
-func (app *VirtAPIApp) Run() {
-	log.Fatal(http.ListenAndServe(app.Address(), nil))
+func (app *virtAPIApp) getClientCert() error {
+	authConfigMap, err := app.virtCli.CoreV1().ConfigMaps(metav1.NamespaceSystem).Get("extension-apiserver-authentication", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	clientCA, ok := authConfigMap.Data["client-ca-file"]
+	if !ok {
+		return fmt.Errorf("client-ca-file value not found in auth config map.")
+	}
+
+	app.clientCABytes = []byte(clientCA)
+	return nil
 }
 
-func (app *VirtAPIApp) AddFlags() {
-	app.InitFlags()
+func (app *virtAPIApp) getSelfSignedCert() error {
+	var ok bool
+	generateCerts := false
+	secret, err := app.virtCli.CoreV1().Secrets(metav1.NamespaceSystem).Get(virtApiCertSecretName, metav1.GetOptions{})
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			generateCerts = true
+		} else {
+			return err
+		}
+	}
 
-	app.BindAddress = defaultHost
-	app.Port = defaultPort
+	if generateCerts {
+		// Generate new certs if secret doesn't already exist
+		caKeyPair, _ := triple.NewCA("kubevirt.io")
+		keyPair, _ := triple.NewServerKeyPair(
+			caKeyPair,
+			"virt-api.kube-system.pod.cluster.local",
+			"virt-api",
+			"kube-system",
```

Comment:
The namespace can be read from `/var/run/secrets/kubernetes.io/serviceaccount/namespace`.

Reply:
fixed. namespace is now dynamically detected